pkg/api/types.go

```diff
@@ -2154,3 +2154,32 @@ type RangeAllocation struct {
 	// a single allocated address (the fifth bit on CIDR 10.0.0.0/8 is 10.0.0.4).
 	Data []byte `json:"data"`
 }
+
+// LockSpec is the specification of a resource lock.
+type LockSpec struct {
```

Comment:
not a bad idea, curious of an example of what kind of metadata  you mean?  If not urgent, maybe in follow on PR... but also could easily be done here (tim/rob will I'm sure have feedback next wk when  back)